src/commands/heroku.rs

```diff
@@ -75,46 +80,48 @@ pub fn get_apps(ctx: &mut Context, msg: &Message, _args: Args) -> CommandResult
 #[command]
 #[num_args(1)]
 pub fn restart_app(ctx: &mut Context, msg: &Message, mut args: Args) -> CommandResult {
-    let config = bot_config(ctx);
-
     let app_name = args
         .single::<String>()
         .expect("You must include an app name");
 
-    let response = heroku_client(&config.heroku_api_key)
-        .delete_empty()
-        .apps()
-        .app_name(&app_name)
-        .app_dynos()
-        .execute::<Value>();
+    let ctx_clone = ctx.clone();
+    let data = ctx_clone.data.read();
+
+    let heroku_client = data
+        .get::<HerokuClient>()
+        .expect("Expected Heroku client")
+        .clone();
+
+    let response = heroku_client.client.request(&dynos::DynoAllRestart {
+        app_id: app_name.clone(),
+    });
+    println!("response: {:?}", response);
 
     msg.reply(
         ctx,
         match response {
-            Ok((_, _, Some(_object))) => format!("All dynos in {} have been restarted.", app_name),
-            Ok((_, _, None)) => "There is no Heroku app by that name".into(),
-            Err(err) => {
-                println!("Err {}", err);
-                "An error occured while fetching your Heroku app".into()
+            Ok(_response) => format!("All dynos in {} have been restarted.", app_name),
+            Err(e) => {
+                println!("Error: {}", e);
+                "An error occured when trying to restart your Heroku app".into()
             }
         },
     )?;
 
     Ok(())
 }
 
-fn heroku_client(api_key: &str) -> heroku_rs::client::Heroku {
-    Heroku::new(api_key).unwrap()
-}
-
-fn app_response(app: HerokuApp) -> String {
+fn app_response(app: heroku_rs::endpoints::apps::App) -> String {
     format!(
         "\nApp ID: {}\nApp Name: {}\nReleased At: {}\nWeb URL: {}\n\n",
-        app.id, app.name, app.released_at, app.web_url
+        app.id,
+        app.name,
+        app.released_at.unwrap(),
```

Comment:
```suggestion
        app.released_at.map(|date| date.as_str()).unwrap_or("never"),
```

This could crash if an app never had any release. Let's write "never" instead?

Reply:
Why would this crash if the app was never released?

Reply:
Ah, sorry, I read _your_ code instead of Nell's. :)

Reply:
Because `released_at` would be `None` and we unwrap it.

Reply:
No worries!

Reply:
As written, this returned an error:

```bash
error[E0515]: cannot return value referencing function parameter `date`
   --> src/commands/heroku.rs:110:55
    |
110 |     let app_released_at =  app.released_at.map(|date| date.as_str()).unwrap_or("never");
    |                                                       ----^^^^^^^^^
    |                                                       |
    |                                                       returns a value referencing data owned by the current function
    |                                                       `date` is borrowed here
```

But I was able to get around it by rewriting it to:

```rust
fn app_response(app: heroku_rs::endpoints::apps::App) -> String {
    format!(
        "\nApp ID: {}\nApp Name: {}\nReleased At: {}\nWeb URL: {}\n\n",
        app.id,
        app.name,
        app.released_at.unwrap_or("never".to_string()),
        app.web_url
    )
}
```

Reply:
Did you push this change?

Reply:
Gah - it's pushed now!